internal/clients/diffreviewer/github/github_client.go

```diff
@@ -14,13 +16,24 @@ type Client struct {
 }
 
 // NewAuthenticatedClient generates an authenticated github client
-func NewAuthenticatedClient(token string) *Client {
+func NewAuthenticatedClient(token string, baseUrl string) *Client {
 	ctx := context.Background()
 	ts := oauth2.StaticTokenSource(
 		&oauth2.Token{AccessToken: token},
 	)
 	tc := oauth2.NewClient(ctx, ts)
 	githubClient := github.NewClient(tc)
+	// for enterprise
+	if baseUrl != "" {
+		u, _ := url.Parse(baseUrl)
+		if !strings.HasSuffix(u.Path, "/") {
```

Comment:
is it not worth just overwriting the path entirely with fmt.String("%v%v", u.HostName, githubAPIPath ). Do we want them to customize this. Seems like we should ensure it's always v3?

Reply:
if they don't specify a protocol, only `u.Path` has a value, the full path including the suffixes, and u.Host is empty. If we want to go the suggested route ^, we'd have to specifically tell users to prepend the protocol when setting their base url.